Lib/ufo2ft/filters/propagateAnchors.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from fontTools.misc.transform import Transform
+from ufo2ft.filters import BaseFilter
+
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class PropagateAnchorsFilter(BaseFilter):
+
+    def set_context(self, font, glyphSet):
+        ctx = super(PropagateAnchorsFilter, self).set_context(font, glyphSet)
+        ctx.processed = set()
+        return ctx
+
+    def __call__(self, font, glyphSet=None):
+        if super(PropagateAnchorsFilter, self).__call__(font, glyphSet):
+            modified = self.context.modified
+            if modified:
+                logger.info('Glyphs with propagated anchors: %i' %
+                            len(modified))
+            return modified
+
+    def filter(self, glyph):
+        if not glyph.components:
+            return False
+        _propagate_glyph_anchors(glyph.font, glyph, self.context.processed)
+        return True
+
+
+def _propagate_glyph_anchors(ufo, parent, processed):
+    """Propagate anchors for a single parent glyph."""
```

Comment:
what you mean exactly by "parent"?
Maybe the docstring should be clearer what this filter actually does. It "propagates anchors", but from where and to where, and how?

Reply:
I’ll rename `parent` to `composite`. The anchors are propagated from all base glyphs to the composite glyphs and to all the glyphs used in between.